pkg/pkiaas/api/ssh_controller.go

```diff
@@ -0,0 +1,281 @@
+		return
+	}
+
+	httpErr = ssh.CreateSSHTemplate(template, storage)
+	if httpErr != (httperror.HTTPError{}) {
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+}
+
+// DeleteSSHTemplateHandler Accepts the HTTP request with the SSH template that is desired to be deleted
+// and passes it to the SSH service
+func DeleteSSHTemplateHandler(w http.ResponseWriter, r *http.Request) {
+	authHeader := r.Header.Get("Authorization")
+
+	// Ensure that the requesting entity can both authenticate to the SSH service, then extract
+	// the template name from the URI and test that it has authorization to access the
+	// Delete Template endpoint for the requested template
+	storage := context.Get(r, "Storage").(backend.Storage)
+	err := storage.GetAccessControl().Authenticate(authHeader)
+	if err != nil {
+		httpErr := httperror.InvalidAuthn(err.Error())
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+
+	templateName := mux.Vars(r)["templateName"]
+
+	err = storage.GetAccessControl().DeleteSSHTemplate(authHeader, templateName)
+	if err != nil {
+		httpErr := httperror.InvalidAuthz(err.Error())
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+
+	httpErr := ssh.DeleteSSHTemplate(templateName, storage)
+	if httpErr != (httperror.HTTPError{}) {
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+	w.WriteHeader(http.StatusNoContent)
+}
+
+//CreateSSHCertificateHandler Accepts the HTTP request with all the required data to create a new SSH certificate
+// and returns the newly created SSH certificate via HTTP response
+func CreateSSHCertificateHandler(w http.ResponseWriter, r *http.Request) {
+	if !pki.ValidateContentType(r.Header, "application/json") {
+		httpErr := httperror.InvalidContentType()
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+
+	// Ensure that the requesting entity can both authenticate to the SSH service, as well as
+	// has authorization to access the Create Certificate endpoint using the requested template
+	storage := context.Get(r, "Storage").(backend.Storage)
+	authHeader := r.Header.Get("Authorization")
+	err := storage.GetAccessControl().Authenticate(authHeader)
+	if err != nil {
+		httpErr := httperror.InvalidAuthn(err.Error())
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+	var certReq types.SSHSignRequest
+	decoder := json.NewDecoder(r.Body)
+	decoder.DisallowUnknownFields()
+	err = decoder.Decode(&certReq)
+	if err != nil {
+		httpErr := httperror.RequestDecodeFail(err.Error())
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+
+	err = storage.GetAccessControl().CreateSSHCertificate(authHeader, certReq.TemplateName)
+	if err != nil {
+		httpErr := httperror.InvalidAuthz(err.Error())
+		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
+		return
+	}
+
+	sshCert, httpErr := ssh.CreateSSHCertificate(certReq, storage)
+	if httpErr != (httperror.HTTPError{}) {
```

Comment:
This doesn't seem right. Shouldn't we having something like:
```
		http.Error(w, httpErr.JSON(), httpErr.HTTPResponse)
                return
```

Reply:
Yep, this is why you're the best 🥇